### DISCUSSION TITLE
Persistent Exercise Notes

### DISCUSSION THREAD

**Discussion by TimKamppinen** [astashov/liftosaur#236] [Ideas]
Need an option to make a note persistent so that it shows up automatically whenever you do that exercise.

I do a a lot of exercises with cables, lever arms, incline bench, etc where I need to write down how I set up the equipment so I can do it quickly each time. Bench on 30 degree incline, cables on position 24, what position to put j cups, spotter arms, etc.

For it to be useful it needs to carry over every time I do the exercise regardless of whether I change my program or if it’s just in an ad hoc workout. And if I come up with a new exercise variation in the middle of a workout I need to be able to quickly add a persistent note so I can do it the same way next time.

Other apps like Strong have this feature and it's great because I can throw in some obscure variation that I haven't done for 6 months and all I need to do is glance at the notes and I can easily set it up the same way without wasting time trying to figure out the angles and settings.

This would be equally valuable for people who work out at a commercial gym where they need to change the settings the same way every time on a machine to fit their body proportions.

One thing that would improve on Strong's implementation would be if the persistent note gets saved as a regular note after the workout is finished. That way you can look back and see historically on each workout how the exercise was set up, but if you decide to change the setup, you can edit the persistent note and it will carry forward without messing up the historical data. By looking in the log you could see when the note changes and know when it was that you changed your setup. 

In Strong, the persistent notes don't show up in the workout history so if you change them they will carry forward but there's no record of what they used to say or when they were changed.

**Comment by yrinkel** [comment]
Could you check if this idea is a duplicate of idea #139 (equipment specific comments)? If yes could you please close this one to reduce clutter?

https://github.com/astashov/liftosaur/discussions/139#discussion-7090882

**Reply by TimKamppinen** [reply]
I saw that one but the way it was written made me think that the note would be tied to the "Equipment" setting of the exercise, like "Barbell" "Dumbbell" "Cable" etc.  I would want to use persistent notes on the exercise itself regardless of what equipment is set for it and even if the equipment is set to "none" since that's how I set up all my custom exercises. 

I find the different "equipment" and "gym" settings to be way over complicated and don't use them, I just set the default rounding directly on each exercise. Just because I do a lat pulldown and a cable lateral raise on the same equipment, doesn't mean I want to use the same rounding preferences/increments for each one.

Maybe I was misreading your ticket and you just meant a note on the exercise that references equipment but isn't tied to the app-specified equipment. In that case I would be fine with these being merged and the votes pooled together.

**Reply by yrinkel** [reply]
Ah the equipment does not set the increment. That’s defined by your progression setting (lp, dp or custom). 

The only thing the equipment setting does is to allow Liftosaur to tell you exactly what to load (plates/barbell combination etc…) on the barbell/machines to achieve your target weight for each set. 

The “gym” setting allows you to define different equipment for different gyms, as they usually have different plates, machines, etc... Then you can simply change gym in the settings and Liftosaur will tell you what to load on the machines (which have different empty weight and plates combinations) to achieve exactly what you normally lift in the other gym.

**Reply by TimKamppinen** [reply]
I think it also affects rounding depending on what plates you have defined in the equipment setting, which can conflict with the rounding set on the exercise according to some recent discussion on the discord. Regardless, I don’t use those settings and need persistent notes that aren’t tied to “equipment”.

**Comment by astashov** [comment]
I think this is done
